challenges/classes.js

```diff
@@ -1,5 +1,42 @@
 // 1. Copy and paste your prototype in here and refactor into class syntax.
 
+//base class
+class CuboidMaker{
+    constructor (attributes){ 
+    this.length = attributes.length;
+    this.width = attributes.width;
+    this.height = attributes.height;
+    }
+    volume(){
+        return (this.length)*(this.width)*(this.height);
+    }
+    surfaceArea(){
+        return 2*((this.length)*(this.width)+(this.length)*(this.height)+(this.width)*(this.height));
+    }
+  }
+ //sub class
+ class CubeMaker extends CuboidMaker{
+     constructor(attributes){
+         super(attributes);
```

Comment:
While you've partially done this stretch goal correctly (great job!), this isn't really a CubeMaker. It's another CuboidMaker. The biggest difference between a Cube and a Cuboid is that all the sides of a Cube are the same length, but a Cuboid can have sides of different length. As such, a CubeMaker would only take one argument, as all of the sides would be equal to that one argument. Here is an example, please let me know if you have any questions: 
```
class CubeMaker extends CuboidMaker {
    constructor(length){
        super(length);
        this.length = length;
        this.width = length;
        this.height = length;
    }
    volume() {
      return Math.pow(this.length, 3);
    }
    surfaceArea() {
      return 6 * Math.pow(this.length, 2);
    }
}
const cube = new CubeMaker(5);
```
Also take note of how the `volume()` and `surfaceArea()` methods are different.